File: languages/java/src/main/java/com/bloock/sdk/entity/identity_v2/StringEnumAttributeDescriptor.java
package com.bloock.sdk.entity.identity_v2;

import com.bloock.sdk.bridge.proto.IdentityEntitiesV2;
import java.util.List;

/**
 * Represents a descriptor for an attribute with a string enum value.
 */
public class StringEnumAttributeDescriptor extends AttributeDescriptor {
  private List<String> enumeration;

  /**
   * Constructs an StringEnumAttributeDescriptor object with the specified parameters.
   * @param displayName
   * @param technicalName
   * @param description
   * @param required
   * @param enumeration
   */
  public StringEnumAttributeDescriptor(
      String displayName,
      String technicalName,
      String description,
      Boolean required,
      List<String> enumeration) {
    super(displayName, technicalName, description, required);
    this.enumeration = enumeration;
  }

  public static StringEnumAttributeDescriptor fromProto(
      IdentityEntitiesV2.StringEnumAttributeDefinitionV2 res) {
    return new StringEnumAttributeDescriptor(
        res.getDisplayName(),
        res.getId(),
        res.getDescription(),
        res.getRequired(),
        res.getEnumList());
  }

  public IdentityEntitiesV2.StringEnumAttributeDefinitionV2 toProto() {
    return IdentityEntitiesV2.StringEnumAttributeDefinitionV2.newBuilder()
        .setDisplayName(this.displayName)
        .setId(this.technicalName)
        .setDescription(this.description)
        .setRequired(this.required)
        .addAllEnum(this.enumeration)
        .build();
  }
}

File: languages/js/src/entity/identity_v2/string_enum_attribute_descriptor.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities_v2";
import { AttributeDescriptor } from "./attribute_descriptor";

export class StringEnumAttributeDescriptor extends AttributeDescriptor {
  enumeration: string[];

  constructor(
    displayName: string,
    technicalName: string,
    description: string,
    required: boolean,
    enumeration: string[]
  ) {
    super(displayName, technicalName, description, required);
    this.enumeration = enumeration;
  }

  public toProto(): identityEntitiesProto.StringEnumAttributeDefinitionV2 {
    return identityEntitiesProto.StringEnumAttributeDefinitionV2.fromPartial({
      displayName: this.displayName,
      id: this.technicalName,
      description: this.description,
      required: this.required,
      enum: this.enumeration
    });
  }

  static fromProto(
    r: identityEntitiesProto.StringEnumAttributeDefinitionV2
  ): StringEnumAttributeDescriptor {
    return new StringEnumAttributeDescriptor(
      r.displayName,
      r.id,
      r.description,
      r.required,
      r.enum
    );
  }
}
